### DISCUSSION TITLE
Multiple GetCollection() declaration in ApiResource

### DISCUSSION THREAD

**Discussion by SimonAmade** [api-platform/api-platform#2402] [Q&A]
Hello, I have multiple GetCollection with each a specific logic, it seems that I can't declare multiple GetCollection like so
`#[ApiResource(
    operations: [
        new Get(),
        new GetCollection(uriTemplate: 'uri/template/one'),
        new GetCollection(uriTemplate: 'uri/template/two')
    ]
)]`
The second returns 'This route does not aim to be called'. I have tried to name each collection. Nothing is working.
The only solution is for me to declare two ApiResource within my entity and one GetCollection in each.
Is it the correct way, I am doing something wrong ?

**Comment by soyuka** [comment]
It may be a bug and yes declaring 2 resources is better. Anyways, it should work. Still, what is your use case this looks wrong semantically.

**Reply by SimonAmade** [reply]
Ok thank you !
What do you mean by this looks wrong semantically ?

**Reply by soyuka** [reply]
In a resource-oriented API having two ways (read URI) of retrieving the same data looks wrong to me. What is the use case?

**Reply by SimonAmade** [reply]
Well, in my case I have a table Activity, each row has a price and a date.
I need a GetCollection that returns let say, the total price, and another one that returns the total by date.
`#[ApiResource( 
operations: [
new GetCollection(uriTemplate: 'activities/total'), 
new GetCollection(uriTemplate: 'activities/totalByDate/') ] 
)]`
I hope I am clear

**Reply by soyuka** [reply]
I see so this is an aggregation of data, IMHO you should probably have two different PHP classes to represent that:

```
namespace App\ApiResource;

#[Get(uriTemplate: '/activites/total', provider: ActivitiesTotalProvider::class, uriVariables: [])]
class ActivitiesTotal {
    public $total;
    /** @var Activites */
    public $activities;
}
``` 

and 

```
namespace App\ApiResource;

#[GetCollection(uriTemplate: '/activites/total/byDates', provider: ActivitiesTotalByDAtesProvider::class)]
class ActivitiesTotalByDates {
    public $nbOfActivities;
    public $total;
    public $month;
}
```

**Reply by SimonAmade** [reply]
I see, but can I still benefit from the filters by doing so ?

**Reply by colesnic89** [reply]
Maybe it is stupid, but it works as expected
```
namespace App\ApiResource;

use App\Entity\ConversationItem;

...

#[ApiResource(
    operations: [
        new GetCollection(
            uriTemplate: '/conversation_items/search',
            normalizationContext: ['groups' => ['ConversationItem:GetCollection']],
            provider: ConversationItemSearchProvider::class,
            parameters: [
                'search' => new QueryParameter(
                    key: 'search',
                    schema: ['type' => 'string'],
                    description: 'Search by message or peerNumber using Elasticsearch'
                ),
            ],
        )
    ]
)]
class ConversationItemSearch extends ConversationItem
{

}
```
